staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -49,7 +51,7 @@ type FeatureSupportChecker interface {
 	// If client A supports and client B doesn't support the feature, the `Supports` will
 	// first return true at client A initializtion and then return false on client B
 	// initialzation, it can flip the support at runtime.
-	CheckClient(ctx context.Context, c client, feature storage.Feature) error
+	CheckClient(ctx context.Context, cancel context.CancelFunc, c client, feature storage.Feature) error
```

Comment:
Cancel should be owned by storage, not feature checker.

Reply:
+1

In general, etcd3.New() should probably return `DestroyFunc` (or cancel or whatever) and that should be included into `DestroyFunc` that is returned from newEtTCD3Storage

Reply:
Thanks for all the reviews @serathius @wojtek-t , really appreciated. Then i think we can use `client.Ctx()` as a context here which will closed on `destroyFunc` automatically. I'll try to make the changes by tomorrow.